cli/azd/extensions/azure.ai.agents/internal/project/service_target_agent.go

```diff
@@ -693,9 +707,17 @@ func (p *AgentServiceTargetProvider) startAgentContainer(
 	// Create agent client
 	agentClient := agent_api.NewAgentClient(azdEnv["AZURE_AI_PROJECT_ENDPOINT"], p.credential)
 
-	// Start agent container (minReplicas and maxReplicas are already int32)
+	var minReplicas, maxReplicas *int32
+	if foundryAgentConfig.Container != nil && foundryAgentConfig.Container.Scale != nil {
+		minReplicasInt32 := int32(foundryAgentConfig.Container.Scale.MinReplicas)
+		maxReplicasInt32 := int32(foundryAgentConfig.Container.Scale.MaxReplicas)
```

Comment:
Make sure that when the values are not defined in azure.yaml, we don't set them as `0` here (they default value).
nit. Validate the cast to int32 in case people add crazy big numbers or negative number or other symbols in the azure.yaml config to prevent panics

Reply:
Since the config objects expect an int, we should fail sooner than this for any "other symbols"